[AIRFLOW-833] Add an `airflow default_config` sub-command

gsakkis opened the pull request:
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- *https://issues.apache.org/jira/browse/AIRFLOW-833*

jlowin commented on airflow/configuration.py:
```diff
@@ -688,18 +680,13 @@ def as_dict(self, display_source=False, display_sensitive=False):
-        self.read_string(parameterized_config(TEST_CONFIG))
-        # then read any "custom" test settings
-        self.read(TEST_CONFIG_FILE)
+    def load_or_save_config_file(self, config_file, config_string=None):
```
This should be split into two methods, `load_config_file` and `save_config_file`. Otherwise it will be extremely confusing to figure out what it's doing when reading code.

gsakkis committed:
[AIRFLOW-833] Add an `airflow default_config` sub-command

- Generate the default airflow.cfg explicitly with a command,
  not when simply importing airflow.
- The default airflow.cfg is also created on `airflow initdb`
  if it doesn't exist already.
- Don't create the unittests.cfg unless unit_test_mode is true.
